content/6_Advanced/Vectorization.mdx

```diff
@@ -133,3 +133,265 @@ instead of faster (ex. see
 <!-- Anyways, keep in mind that these rarely make a difference in competitive programming. -->
 
 <!-- <IncompleteSection /> -->
+
+## Example - Ligatures
+
+[Problem](https://open.kattis.com/problems/ligatures?editresubmit=13798177)
+We have yet to find a problem on this topic. In the meantime, use this.
+
+The problem involves counting the occurrences of specific pairs of characters (ligatures) in a given text corpus based on multiple sets of suggested ligatures. The primary challenge is efficiently handling large input sizes and ensuring that the counting process respects the rule that ligatures can't overlap.
+
+### Solution
+
+This solution to the problem uses SIMD (Single Instruction, Multiple Data) parallelism to efficiently count the occurrences of suggested ligatures within a large corpus:
+
+**Overview**
+1. **Preprocessing the Corpus**:
+- The corpus is preprocessed to identify all pairs of consecutive characters, which are stored in a pairs vector.
+
+2. **Setting Up Bitmasks**:
+- A bitmask array active is set up where each bit represents whether a specific ligature is part of a query.
+The bitmask for each query is configured such that each suggested ligature toggles a specific bit.
+
+3. **Counting Occurrences Using SIMD**:
+- The core of the solution utilizes SIMD instructions to parallelize the counting process.
+The program processes the corpus in chunks, accumulating counts for each query simultaneously using AVX2 intrinsics.
+
+**Time Complexity:** $\mathcal{O}(N + Q * K)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#pragma GCC target("avx2")
+#include <immintrin.h>
+#include <iostream>
+#include <string>
+#include <vector>
+using namespace std;
+
+#define rep(i, from, to) for (int i = from; i < (to); ++i)
+#define trav(a, x) for (auto &a : x)
+#define all(x) x.begin(), x.end()
+#define sz(x) (int)(x).size()
+typedef long long ll;
+typedef pair<int, int> pii;
+typedef vector<int> vi;
+
+const int SIMD_BITS = 256;
+const int MAXQ = 100'000;
+
+typedef __m256i mi;
+
+void set_bit(mi &m, size_t ind) {
+	union {
+		mi m;
+		uint32_t ar[SIMD_BITS / 32];
+	} u;
+	u.m = m;
+	u.ar[ind / 32] = (uint32_t)(u.ar[ind / 32] | (1 << (ind % 32)));
+	m = u.m;
+}
+
+void printm(mi m) {
+	union {
+		mi m;
+		uint32_t ar[SIMD_BITS / 32];
+	} u;
+	u.m = m;
+	rep(i, 0, SIMD_BITS) { cout << ((u.ar[i / 32] >> (i % 32)) & 1); }
+	cout << endl;
+}
+
+mi active[MAXQ / SIMD_BITS + 1][26 * 26 + 1]{};  // ~26*26 * MAXQ / 8 bytes
+
+int main() {
+	cin.sync_with_stdio(false);
+	cin.exceptions(cin.failbit);
```

Comment:
what's this

Reply:
This code snippet sets up the environment for using SIMD (Single Instruction, Multiple Data) operations with AVX2 instructions in C++. 
The function _set_bit_ sets a specific bit in a 256-bit vector (_mi_ type) at the given index (_ind_). The function _printm_  prints the binary representation of a 256-bit vector (_mi_ type).
In the main function, _cin.sync_with_stdio(false)_  disables the synchronization between C++ standard streams and C standard streams, which improves I/O performance. _cin.exceptions(cin.failbit)_ configures the input stream to throw exceptions if it encounters an input failure, enabling more robust error handling.